src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+import * as yup from 'yup';
+import { toast } from 'react-toastify';
+import { createYupTestFromQuery } from '~utils/yup/tests';
+
+import { GetProfileByEmailDocument, useUpdateUserProfileMutation } from '~gql';
+import { UserPreferencesFormProps } from './types';
+import { useAppContext, useCanEditProfile } from '~hooks';
+import Toast from '~shared/Extensions/Toast';
+
+export const useUserPreferencesPage = () => {
+  const { formatMessage } = useIntl();
+  const { updateUser } = useAppContext();
+  const [editUser] = useUpdateUserProfileMutation();
+  const { user } = useCanEditProfile();
+
+  const EMAIL_REGEX = /^[\w-.]+@([\w-]+\.)+[\w-]+$/;
```

Comment:
please export and use regex from here: src/components/common/CreateUserWizard/validation.ts